### PR TITLE
New config, :warn_test_pattern

### PR DESCRIPTION
**Pull request by gustf**
Display warning for files matching :warn_test_pattern that are not
matching :test_pattern. Default config,"*_test.ex".

### CODE REVIEW

**Comment by paulcsmith** [comment]
Hey @gustf. I didn't have a chance to look through the code and test it out on my machine, but one thing you might want to do is add documentation for the new config option. You can see near the top of the file there is documentation for stuff like `test_pattern`. You could add something similar for this new option

**Comment by gustf** [comment]
Hi again @paulcsmith. Good point about the documentation. Added some, also built the doc with ex_doc and checked that it looked ok. Nice auto-build-documentation tool! :)

File: lib/mix/lib/mix/tasks/test.ex [hunk 1]
```diff
@@ -186,8 +189,15 @@ defmodule Mix.Tasks.Test do
     # Finally parse, require and load the files
     test_files   = parse_files(files, test_paths)
     test_pattern = project[:test_pattern] || "*_test.exs"
-
-    case Mix.Utils.extract_files(test_files, test_pattern) do
+    warn_test_pattern = project[:warn_test_pattern] || "*_test.ex"
+
+    matched_test_files = Mix.Utils.extract_files(test_files, test_pattern)
+    matched_warn_test_files = Mix.Utils.extract_files(test_files, warn_test_pattern)
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/test.ex]: `Mix.Utils.extract_files(test_files, warn_test_pattern) -- matched_test_files` is equivalent to the "reject with member?" call.

**Comment by josevalim** [comment]
This looks great. I have added just one tiny comment.

File: lib/mix/lib/mix/tasks/test.ex [hunk 2]
```diff
@@ -203,6 +213,12 @@ defmodule Mix.Tasks.Test do
     end
   end
 
+  defp display_warn_test_pattern(files, pattern) do
+    for file <- files do
+      Mix.shell.info "Warning: #{file} is not matching \"#{pattern}\" and won't be run. "
```
Comment (paulcsmith) [inline lib/mix/lib/mix/tasks/test.ex]: What do you think of changing "is not matching" to "does not match"?

**Comment by gustf** [comment]
@josevalim @paulcsmith Thanks and I agree with both of you, added a new commit with those changes.

**Comment by paulcsmith** [comment]
@gustf You're welcome and great job on this!

**Comment by lexmag** [comment]
:+1:. @gustf could you squash everything into one commit?

File: lib/mix/lib/mix/tasks/test.ex [hunk 3]
```diff
@@ -203,6 +212,12 @@ defmodule Mix.Tasks.Test do
     end
   end
 
+  defp display_warn_test_pattern(files, pattern) do
+    for file <- files do
+      Mix.shell.info "Warning: #{file} does not match \"#{pattern}\" and won't be run."
```
Comment (eksperimental) [inline lib/mix/lib/mix/tasks/test.ex]: Messeages shouldnt have a trailing period

**Comment by gustf** [comment]
@eksperimental :+1: 
@lexmag Ok, done (I think, it looks like its ok but this is the first time I use squash.) :)

File: lib/mix/lib/mix/tasks/test.ex [hunk 4]
```diff
@@ -108,6 +108,9 @@ defmodule Mix.Tasks.Test do
 
     * `:test_pattern` - a pattern to load test files, defaults to `*_test.exs`.
 
+    * `:warn_test_pattern` - a pattern to match potentially missed test files 
+       and display a warning, defaults to `*_test.ex`.
```
Comment (eksperimental) [inline lib/mix/lib/mix/tasks/test.ex]: This will break list, please use two spaces only (there are three here)

**Comment by gustf** [comment]
:+1: @eksperimental  fixed

**Comment by eksperimental** [comment]
sweet :sweet_potato:

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**